### PR TITLE
Support for new CIS Microsoft Azure Foundations Benchmark v1.5.0 Closes #126

### LINKED ISSUE #126: Support for new CIS Microsoft Azure Foundations Benchmark v1.5.0
In August 2022, CIS released a new version of their Azure benchmark:

- CIS Microsoft Azure Foundations Benchmark v1.5.0

I was surprised to see that the newly released CIS AWS 1.5.0 benchmark is already integrated in the AWS Compliance Mod.
Do you plan to integrate also the new CIS benchmark in the Azure Compliance Mod?

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist
- [ ] Issue(s) linked

### CODE REVIEW

File: cis_v150/docs/cis_v150_2_5.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+## Description
+
+Ensure that the latest OS patches for all virtual machines are applied.
+
+Windows and Linux virtual machines should be kept updated to:
+  - Address a specific bug or flaw
+  -  Improve an OS or application’s general stability
```
Comment (misraved) [inline cis_v150/docs/cis_v150_2_5.md:7]:
```suggestion
  - Improve an OS or application’s general stability
```

File: cis_v150/docs/cis_v150_5_2_3.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    - `All network securiy groups or `< your network security group >`
+    - `< Resource Name >` - The subscription, group, or resource you selected
+8. Click `Done`
+9. Under the Condition tab, click `Add Condition` (the `Select a signal` window may automatically open without clicking)
+10. In the `Select a signal` window, under the "Signal Name" heading, click Create or Update Network Security Group (Microsoft.Network/networkSecurityGroups)
+11. Under the Actions tab, choose appropriately:
+    - Select action groups - If you have an existing action group to notify the necessary personnel.
+    - Create action group - If you do not have an existing action group or want to create a new one.
+12. Under the Details tab, fill in:
+    -  Resource group - Select the resource group you want the alert rule to reside in.
```
Comment (misraved) [inline cis_v150/docs/cis_v150_5_2_3.md]:
```suggestion
    - Resource group - Select the resource group you want the alert rule to reside in.
```

File: cis_v150/docs/cis_v150_5_2_4.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+Create an activity log alert for the Delete Network Security Group event.
+
+Monitoring for "Delete Network Security Group" events gives insight into network access changes and may reduce the time it takes to detect suspicious activity.
+
+## Remediation
+
+### From Azure Portal
+
+  1. Navigate to the `Monitor` blade
```
Comment (misraved) [inline cis_v150/docs/cis_v150_5_2_4.md]:
```suggestion
 1. Navigate to the `Monitor` blade
```

File: cis_v150/docs/cis_v150_9_10.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+## Remediation
+
+### From Azure Portal
+
+1. Go to the Azure Portal
+2. Select `App Services`
+3. Click on an app
+4. Select `Settings` and then `Configuration`
+5. Under `General Settings`, for the `Platform Settings`, the `FTP state should be set to `Disabled` or `FTPS Only`
```
Comment (misraved) [inline cis_v150/docs/cis_v150_9_10.md]:
```suggestion
5. Under `General Settings`, for the `Platform Settings`, the `FTP state` should be set to `Disabled` or `FTPS Only`
```

File: cis_v150/section_1.sp [hunk 5]
```diff
@@ -0,0 +1,547 @@
...
+    cis_item_id = "1.14"
+    cis_level   = "1"
+    cis_type    = "manual"
+    service     = "Azure/ActiveDirectory"
+  })
+}
+
+control "cis_v150_1_15" {
+  title         = "1.15 Ensure That 'Guest users access restrictions' is set to 'Guest user access is restricted to properties and memberships of their own directory objects'"
+  description   = "Limit guest user permissions.."
```
Comment (misraved) [inline cis_v150/section_1.sp:397]:
```suggestion
  description   = "Limit guest user permissions."
```

File: cis_v150/section_4.sp [hunk 6]
```diff
@@ -0,0 +1,472 @@
...
+    "cis_item_id" = "4.2.1"
+    "cis_level"   = "2"
+    "cis_type"    = "automated"
+    service       = "Azure/SQL"
+  })
+}
+
+control "cis_v150_4_2_2" {
+  title         = "4.2.2 Ensure that Vulnerability Assessment (VA) is enabled on a SQL server by setting a Storage Account"
+  description   = "EEnable Vulnerability Assessment (VA) service scans for critical SQL servers and corresponding SQL databases."
```
Comment (misraved) [inline cis_v150/section_4.sp:176]:
```suggestion
  description   = "Enable Vulnerability Assessment (VA) service scans for critical SQL servers and corresponding SQL databases."
```

File: cis_v150/section_4.sp [hunk 7]
```diff
@@ -0,0 +1,472 @@
...
+  sql           = query.postgres_db_server_log_disconnections_on.sql
+  documentation = file("./cis_v150/docs/cis_v150_4_3_4.md")
+
+  tags = merge(local.cis_v150_4_3_common_tags, {
+    cis_item_id = "4.3.4"
+    cis_level   = "1"
+    cis_type    = "automated"
+    service     = "Azure/PostgreSQL"
+  })
+}
```
Comment (misraved) [inline cis_v150/section_4.sp:303]: The title and the description of both the controls look similar, please take a look.

File: cis_v150/section_4.sp [hunk 8]
```diff
@@ -0,0 +1,472 @@
...
+
+  tags = merge(local.cis_v150_4_4_common_tags, {
+    type    = "Benchmark"
+    service = "Azure/SQL"
+  })
+}
+
+control "cis_v150_4_4_1" {
+  title         = "4.4.1 Ensure 'Enforce SSL connection' is set to 'Enabled' for Standard MySQL Database Server"
+  description   = "Enable SSL connection on MYSQL Servers."
```
Comment (misraved) [inline cis_v150/section_4.sp:379]:
```suggestion
  description   = "Enable SSL connection on MySQL Servers."
```

File: cis_v150/section_8.sp [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+    cis_item_id = "8.2"
+    cis_level   = "1"
+    cis_type    = "automated"
+    service     = "Azure/KeyVault"
+  })
+}
+
+control "cis_v150_8_3" {
+  title         = "8.3 Ensure that the Expiration Date is set for all Secrets in RBAC Key Vaults"
+  description   = "Ensure that all Secrets in Role Based Access Control (RBAC) Azure Key Vaults have an expiration time set"
```
Comment (misraved) [inline cis_v150/section_8.sp:56]:
```suggestion
  description   = "Ensure that all Secrets in Role Based Access Control (RBAC) Azure Key Vaults have an expiration time set."
```

File: cis_v150/section_8.sp [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (misraved) [inline cis_v150/section_8.sp:149]:
```suggestion

```

**Review by misraved — COMMENTED** [review]
make changes as per the review comments

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**